mcp/mcp-cli-api/src/main/java/software/amazon/smithy/java/mcp/cli/ConfigUtils.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.mcp.cli;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import software.amazon.smithy.java.io.ByteBufferUtils;
+import software.amazon.smithy.java.json.JsonCodec;
+import software.amazon.smithy.java.mcp.cli.model.Config;
+import software.amazon.smithy.java.mcp.cli.model.ToolBundleConfig;
+
+/**
+ * Utility class for managing Smithy MCP configuration files.
+ * <p>
+ * This class provides methods for loading, creating, updating, and serializing MCP configurations.
+ */
+public class ConfigUtils {
+
+    private ConfigUtils() {}
+
+    private static final JsonCodec JSON_CODEC = JsonCodec.builder().build();
+
+    /**
+     * Gets the path to the config file.
+     *
+     * @return The path to the config file
+     */
+    private static Path getConfigPath() {
+        String userHome = System.getProperty("user.home");
+        Path configDir = Paths.get(userHome, ".config", "smithy-mcp");
+        return configDir.resolve("config.json");
+    }
+
+    /**
+     * Ensures the config directory exists.
+     *
+     * @throws IOException If there's an error creating directories
+     */
+    private static void ensureConfigDirExists() throws IOException {
+        Path configDir = getConfigPath();
+        if (!Files.exists(configDir)) {
+            Files.createDirectories(configDir);
+        }
+    }
+
+    /**
+     * Loads an existing config file or creates one if it doesn't exist.
+     *
+     * @return The config file
+     * @throws IOException If there's an error creating directories or the file
+     */
+    public static Config loadOrCreateConfig() throws IOException {
+        Path configFile = getConfigPath();
+        ensureConfigDirExists();
+
+        // Check if the config file exists, create it if it doesn't
+        var file = configFile.toFile();
+        if (!file.exists()) {
+            // Create an empty JSON object as the default config
+            try (var writer = new FileWriter(file, StandardCharsets.UTF_8)) {
+                writer.write("{}");
+            }
```

Comment:
Use `Files.writeString(configFile, "{}", StandardOpenOption.CREATE_NEW)` so we don't accidentally clobber a config file that's being written in parallel